src/tests

```diff
@@ -43,7 +43,7 @@ p({"class":"foam.nanos.test.Test","passed":10,"failed":0,"id":"FoldAndReduceTest
hat group.\");\n    test(passed, TEST_MESSAGE);\n    if ( ! passed ) {\n      print(\"Error message mismatch. Actual was: \" + e.getMessage());\n    }\n  } finally {\n    cleanUp(x);\n  }\n}\n\n\n// Try to remove a permission from a group without permission.\nremovePermissionFromGroupWithoutPermissionThrows() {\n  logger_.info(new Object[]{\"removePermissionFromGroupWithoutPermissionThrows\"});\n  TEST_MESSAGE = \"Cannot remove permissions from groups without the appropriate permissions.\";\n\n  // Create a test user.\n  List permissionIds = new ArrayList();\n  X userContext = generateTestUser(permissionIds);\n\n  try {\n    // Try to remove \"*\" from \"admin\".\n    GroupPermissionJunction j = new GroupPermissionJunction.Builder(x)\n      .setSourceId(\"admin\")\n      .setTargetId(\"*\")\n      .build();\n    groupPermissionJunctionDAO.inX(userContext).remove(j);\n\n    // If the put didn't throw, then this test failed.\n    test(false, TEST_MESSAGE);\n  } catch (AuthorizationException e) {\n    passed = e.getMessage().equals(\"You do not have permission to update that group.\");\n    test(passed, TEST_MESSAGE);\n    if ( ! passed ) {\n      print(\"Error message mismatch. Actual was: \" + e.getMessage());\n    }\n  } finally {\n    cleanUp(x);\n  }\n\n  // Do it a second time but with permission to update the group this time, but\n  // not the permission we're trying to remove. You should need both.\n\n  // Create a test user.\n  permissionIds = new ArrayList();\n  permissionIds.add(\"group.update.admin\");\n  userContext = generateTestUser(permissionIds);\n\n  try {\n    // Try to remove \"*\" from \"admin\".\n    j = new GroupPermissionJunction.Builder(x)\n      .setSourceId(\"admin\")\n      .setTargetId(\"*\")\n      .build();\n    groupPermissionJunctionDAO.inX(userContext).remove(j);\n\n    // If the put didn't throw, then this test failed.\n    test(false, TEST_MESSAGE);\n  } catch (AuthorizationException e) {\n    passed = e.getMessage().equals(\"Permission denied. You cannot add or remove a permission that you do not have.\");\n    test(passed, TEST_MESSAGE);\n    if ( ! passed ) {\n      print(\"Error message mismatch. Actual was: \" + e.getMessage());\n    }\n  } finally {\n    cleanUp(x);\n  }\n}\n\n\n// Run the tests.\ntry {\n  createGroupWithoutPermissionThrows();\n  addNewPermissionThrows();\n  addGroupWithAdminParentThrows();\n  updateGroupWithoutPermissionThrows();\n  updateGroupParentToAdminThrows();\n  createUserInAdminGroupThrows();\n  updateUserGroupToAdminThrows();\n  updateOwnGroupThrows();\n  createPermissionWithoutPermissionThrows();\n  createPermissionWithPermission();\n  updatePermissionWithoutPermissionThrows();\n  deletePermissionWithoutPermissionThrows();\n  deleteGroupWithoutPermissionThrows();\n  removePermissionFromGroupWithoutPermissionThrows();\n} catch (Throwable e) {\n  e.printStackTrace();\n  test(false, \"An unexpected exception was thrown. Some tests might not have been executed.\");\n}\n","output":""})
```

Comment:
@jlhughes Can you please paste the updated script as a reply to this comment so I can diff it with the current one?